src/Tools/DynamoShapeManager/Utilities.cs

```diff
@@ -392,8 +394,14 @@ public static string GetGeometryFactoryPath2(string rootFolder, Version version)
             }
             //IMPORTANT_ Going forward libg folders will be named as follows: libg_major_minor_build - in reference to ASM.
 
-            var libGFolderName = string.Format("libg_{0}_{1}_{2}", version.Major, version.Minor, version.Build);
-            var libGFolder = Path.Combine(rootFolder, libGFolderName);
+            if (version == null)
+            {
+                throw new ArgumentNullException("version");
+            }
+
+            //lookup libG with a fallback to older versions which share the major version number.
+            var libGFolder = Utilities.GetLibGPreloaderLocation(version, rootFolder);
```

Comment:
With the current implementation of `GetGeometryFactoryPath2` I feel that we can remove the preloadASM() function in D4R since the default call of `GetGeometryFactoryPath` will make the fall back as well, what do you think?

Reply:
yes, I think their implementation can go back to how it was previously.